src/main/java/seedu/address/model/person/ClientTypeContainsKeywordsPredicate.java

```diff
@@ -20,8 +20,14 @@ public ClientTypeContainsKeywordsPredicate(List<String> keywords) {
     @Override
     public boolean test(Person person) {
         Set<ClientType> personClientTypes = person.getClientTypes();
+
+        // If the client type list is empty, return false -> no client type to match (Invalid Client Type)
+        if (this.keywords.isEmpty()) {
+            return false;
+        }
+
         return keywords.stream()
-                .anyMatch(keyword -> personClientTypes.stream().anyMatch(clientType ->
+                .allMatch(keyword -> personClientTypes.stream().anyMatch(clientType ->
```

Comment:
Good change from any to all to ensure AND search and not OR search to ensure proper code usage